lib/ts/recipe/anomalydetection/getClientIp.ts

```diff
@@ -0,0 +1,130 @@
+/**
```

Comment:
how can we test this to make sure we get the ip correctly?

Reply:
I can add some tests around this